pkg/onecli/credentials_test.go

```
@@ -0,0 +1,89 @@
+	"net/http/httptest"
+	"testing"
+)
+
+func TestCredentialProvider_Success(t *testing.T) {
+	t.Parallel()
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/api/user/api-key" {
+			t.Errorf("unexpected path: %s", r.URL.Path)
+		}
+		json.NewEncoder(w).Encode(map[string]string{"apiKey": "oc_test_key_123"})
```

golangci-lint (2.11.4)

[error] Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)